File: back-end/PHP/yii/backend/models/Article.php
<?php

namespace app\models;

use Yii;
use yii\behaviors\AttributeTypecastBehavior;

/**
 * This is the model class for table "article".
 *
 * @property int $id
 * @property string $title 文章标题
 * @property string $author 文章作者
 * @property string $content_html 文章html内容
 * @property string $content_md 文章markdown内容
 * @property string $keywords 文章关键词
 * @property string $description 文章描述
 * @property string $thumb 文章缩略图
 * @property int $is_show 是否显示,1是0否
 * @property int $is_original 是否原创,1是0否
 * @property int $is_top 是否置顶,1是0否
 * @property int $click 文章点击次数
 * @property int $category_id 文章分类ID
 * @property int $deleted_at 删除时间
 * @property int $created_at 创建时间
 * @property int $updated_at 修改时间
 */
class Article extends \yii\db\ActiveRecord
{
    /**
     * {@inheritdoc}
     */
    public static function tableName()
    {
        return 'article';
    }

    public function behaviors()
    {
        return [
            'typecast' => [
                'class' => AttributeTypecastBehavior::class,
                'attributeTypes' => [
                    'created_at' => function ($value) {
                        return date('Y-m-d H:i:s', $value);
                    },
                    'updated_at' => function ($value) {
                        return date('Y-m-d H:i:s', $value);
                    }
                ],
                'typecastAfterValidate' => true,
                'typecastBeforeSave' => false,
                'typecastAfterFind' => true,
            ]
        ];
    }
    /**
     * {@inheritdoc}
     */
    public function rules()
    {
        return [
            [['title', 'author', 'content_html', 'keywords', 'description', 'thumb', 'category_id'], 'required'],
            [['content_html', 'content_md'], 'string'],
            [['is_show', 'is_original', 'is_top', 'click', 'category_id', 'deleted_at', 'created_at', 'updated_at'], 'integer'],
            [['title'], 'string', 'max' => 100],
            [['author'], 'string', 'max' => 20],
            [['keywords', 'description', 'thumb'], 'string', 'max' => 255],
        ];
    }
}

File: back-end/Go/fiber/internal/models/article.go
package models

import (
	"github.com/ohdata/blog/tools/util"
)

const ARTICLE_VIEW = "blog_article_view"

type Article struct {
	ID            *int64         `json:"id" validate:"gt=0"`
	Title         string         `json:"title" validate:"gte=2,lte=100"`
	Author        string         `json:"author" validate:"gte=2,lte=20"`
	Keywords      string         `json:"keywords" validate:"gte=2,lte=255"`
	Description   string         `json:"description" validate:"lte=255"`
	Thumb         string         `json:"thumb" validate:"url"`
	ContentHtml   string         `json:"content_html" validate:"gte=2" mapstructure:"content_html"`
	ContentMd     string         `json:"content_md" gorm:"column:content_md" validate:"gte=2" mapstructure:"content_md"`
	IsShow        *int8          `json:"is_show" validate:"oneof=0 1" mapstructure:"is_show"`
	IsTop         *int8          `json:"is_top" validate:"oneof=0 1" mapstructure:"is_top"`
	IsOriginal    *int8          `json:"is_original" validate:"oneof=0 1" mapstructure:"is_original"`
	Click         int64          `json:"click"`
	CategoryID    *int64         `json:"category_id" validate:"gt=0,numeric" mapstructure:"category_id"`
	CreatedUnix   int64          `json:"created_unix" gorm:"column:created_at"`
	UpdatedUnix   int64          `json:"updated_unix" gorm:"column:updated_at"`
	CreatedAt     string         `json:"created_at" gorm:"-"`
	UpdatedAt     string         `json:"updated_at" gorm:"-"`
	Category      simpleCategory `json:"category" gorm:"-"`
	Tags          []simpleTag    `json:"tags" gorm:"-"`
	CommentsCount int64          `json:"comments_count" gorm:"-"`
}

type SimpleArticle struct {
	ID    int64  `json:"id"`
	Title string `json:"title"`
	Thumb string `json:"thumb"`
}

type simpleCategory struct {
	ID    *int64  `json:"id"`
	Title *string `json:"title"`
}

type simpleTag struct {
	ID        *int64 `json:"id"`
	Name      string `json:"name"`
	ArticleID int64  `json:"-"`
}

func (a *Article) AfterFind() {
	a.CreatedAt = util.GetDateTime(a.CreatedUnix, util.YMDHIS)
	a.UpdatedAt = util.GetDateTime(a.UpdatedUnix, util.YMDHIS)
}
